zlib.h

```diff
@@ -109,8 +109,9 @@ typedef struct z_stream_s {
     int                   data_type;  /* best guess about the data type: binary or text
                                          for deflate, or the decoding state for inflate */
     uint32_t              adler;      /* Adler-32 or CRC-32 value of the uncompressed data */
-#if defined(ZLIB_COMPAT) && defined(X86_64)
-    uint32_t              padding;    /* pad out to the same size as the zlib struct */
+#if defined(ZLIB_COMPAT) && defined(IS_64BIT_LONG)
+    uint32_t              padding;    /* pad out to the same size as the zlib struct: padding
+                                         is needed when the size of long int is 64 bits */
```

Comment:
This code was incorrect originally and will be removed very soon. Assumes little-endian order which isn't true for all supported processors.